datalad_fuse/fsspec_head.py

```diff
@@ -0,0 +1,80 @@
+from itertools import islice
+import sys
+from typing import Any, Dict, Iterator, Optional
+
+from datalad.distribution.dataset import (
+    Dataset,
+    EnsureDataset,
+    datasetmethod,
+    require_dataset,
+)
+from datalad.interface.base import Interface, build_doc
+from datalad.interface.results import get_status_dict
+from datalad.interface.utils import eval_results
+from datalad.support.constraints import EnsureInt, EnsureNone, EnsureStr
+from datalad.support.param import Parameter
+
+from .fsspec import FsspecAdapter
+
+DEFAULT_LINES = 10
+
+
+@build_doc
+class FsspecHead(Interface):
+    """
+    Show leading lines/bytes of an annexed file by fetching its data from a
+    remote URL
+    """
+
+    result_renderer = 'tailored'
+
+    _params_ = {
+        "dataset": Parameter(
+            args=("-d", "--dataset"),
+            doc="""dataset to operate on.  If no dataset is given, an
+            attempt is made to identify the dataset based on the current
+            working directory.""",
+            constraints=EnsureDataset() | EnsureNone(),
+        ),
+        "lines": Parameter(
+            args=("-n", "--lines"),
+            doc="How many lines to show (default 10)",
+            constraints=EnsureInt() | EnsureNone(),
+        ),
+        "bytes": Parameter(
+            args=("-c", "--bytes"),
+            doc="How many bytes to show",
+            constraints=EnsureInt() | EnsureNone(),
+        ),
+        "path": Parameter(
+            args=("path",),
+            doc="Path to an annexed file to show the leading contents of",
+            constraints=EnsureStr(),
+        ),
+    }
+
+    @staticmethod
+    @datasetmethod(name="fsspec_head")
+    @eval_results
+    def __call__(
+        path: str,
+        dataset: Optional[Dataset] = None,
+        lines: Optional[int] = None,
+        bytes: Optional[int] = None,
+    ) -> Iterator[Dict[str, Any]]:
+        ds = require_dataset(dataset, purpose="fetch file data", check_installed=True)
+        if lines is not None and bytes is not None:
+            raise ValueError("'lines' and 'bytes' are mutually exclusive")
+        elif lines is None and bytes is None:
+            lines = DEFAULT_LINES
+        fsa = FsspecAdapter(ds.path)
+        with fsa.open(path) as fp:
```

Comment:
please here convert relative paths to be absolute (or relative to the ds.path) since 

```shell
(git-annex)lena:…bide/Derivatives/ccs_nofilt_global[master]sub-MaxMuna0051366/ses-1/func
$> datalad fsspec-head -n 1 sub-MaxMuna0051366_ses-1_variant-CC200_timeseries.1D
[ERROR  ] FileNotFoundError([Errno 2] No such file or directory: 'sub-MaxMuna0051366_ses-1_variant-CC200_timeseries.1D') (FileNotFoundError) 
(dev3) 1 15692 ->1.....................................:Thu 16 Sep 2021 02:50:12 PM EDT:.
(git-annex)lena:…bide/Derivatives/ccs_nofilt_global[master]sub-MaxMuna0051366/ses-1/func
$> datalad fsspec-head -n 1 $PWD/sub-MaxMuna0051366_ses-1_variant-CC200_timeseries.1D
#1	#2	#3	#4...
```

I will later harmonize (forgot at which stage it is happening) to how datalad does it based on either a `Dataset` or just a `str` (e.g. passed from CLI)

Reply:
Done.

Reply:
awesome, thank you